api/staged_products_property_collection_guid_assigner.go

```diff
@@ -79,20 +79,39 @@ type responsePropertyCollectionItem struct {
 	Data map[interface{}]interface{}
 }
 
-func parseResponsePropertyCollection(rawItems interface{}) (responsePropertyCollection, error) {
+func parseResponsePropertyCollection(input associateExistingCollectionGUIDsInput) (responsePropertyCollection, error) {
 	var collection responsePropertyCollection
 
-	rawItemSlice, ok := rawItems.([]interface{})
+	rawItemSlice, ok := input.ExistingProperty.Value.([]interface{})
 	if !ok {
-		return nil, fmt.Errorf("parseResponsePropertyCollection: failed to convert %v to []interface{}", rawItems)
+		return nil, fmt.Errorf("parseResponsePropertyCollection: failed to convert %v to []interface{}", input.ExistingProperty.Value)
 	}
 
-	for _, item := range rawItemSlice {
+	for index, item := range rawItemSlice {
 		itemMap, ok := item.(map[interface{}]interface{})
 		if !ok {
 			return nil, fmt.Errorf("parseResponsePropertyCollection: failed to convert %v to map[interface{}]interface{}", item)
 		}
 
+		for collectionItemKey, collectionItemObj := range itemMap {
+			collectionItemObjAsMap := collectionItemObj.(map[interface{}]interface{})
+			isCredential := collectionItemObjAsMap["credential"].(bool)
+			if !isCredential {
+				continue
+			}
+
+			credentialName := fmt.Sprintf("%s[%d].%s", input.PropertyName, index, collectionItemKey)
+			apiOutput, err := input.APIService.GetDeployedProductCredential(GetDeployedProductCredentialInput{
+				DeployedGUID:        input.ProductGUID,
+				CredentialReference: credentialName,
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			collectionItemObjAsMap["value"] = apiOutput.Credential.Value
+		}
```

Comment:
All the new logic is here

Reply:
Fetching the credential values like this increases the HTTP requests made by  `om configure-product  --config example-product.config.yml`  from 40 to 60

Reply:
is it going to be 60 requests every time, or just for collection objects?

Reply:
not too worried about the time addition